java-components/cache/src/main/java/com/redhat/hacbs/artifactcache/artifactwatch/ScmLookup.java

```diff
@@ -64,11 +64,11 @@ public void onAdd(ArtifactBuild newObj) {
                             || Objects.equals(newObj.getStatus().getState(), ArtifactBuild.NEW)) {
                         try {
                             if (newObj.getMetadata().getAnnotations() != null
-                                    && newObj.getMetadata().getAnnotations().containsKey(REBUILD)) {
+                                    && newObj.getMetadata().getAnnotations().containsKey(ModelConstants.REBUILD)) {
                                 //if this is a forced rebuild we always update the SCM info
                                 //there is a good chance there may be a new recipe
                                 recipeManager.forceUpdate();
-                                newObj.getMetadata().getAnnotations().remove(REBUILD);
+                                newObj.getMetadata().getAnnotations().remove(ModelConstants.REBUILD);
                                 client.resource(newObj).patch();
```

Comment:
Good point about this not being consistent. We should probably return here after this change, and it will get picked up on the next invocation.